src/compiler/checker.ts

```diff
@@ -7355,6 +7356,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             }
             const parameterTypeNode = serializeTypeForDeclaration(context, parameterType, parameterSymbol, context.enclosingDeclaration, privateSymbolVisitor, bundledImports);
 
+            let { initializer } = parameterDeclaration as ParameterDeclaration;
+            if (parameterDeclaration) initializer ??= getJSDocInitializerParameter(parameterDeclaration as ParameterDeclaration);

```

Comment:
Hm. It seems I shouldn't do this cast, I can't figure out in what case parameterDeclaration is JSDocParameterTag. Will check tomorrow

Reply:
Also it seems logical that default parameter hint from js doc should take priority over node's initializer

As I understand js doc initializer isn't necessary an expression, it can be an arbitrary comment.

Reply:
Okay I got it:

```ts
/**
 * This callback type is called `requestCallback` and is displayed as a global symbol.
 *
 * @callback requestCallback
 * @param {number} responseCode
 * @param {string} responseMessage
 */

/**
 * Does something asynchronously and executes the callback on completion.
 * @param {requestCallback} cb - The callback that handles the response.
 */
 function doSomethingAsynchronously(cb) {
    cb(/*trigger*/)
};
```
And obviously it can't be random comment, content would be stripped after first whitespace (presumable identifier)